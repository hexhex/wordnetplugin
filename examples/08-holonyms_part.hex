%% this returns all (part) holonyms of a word
%% POS: nouns

%% the search number for (part) holonyms
%% is 08.
%% Note: In the WordNet documentation this is
%%       the search number for part meronyms.
%%       In fact search 08 fetches words, for
%%       which *word* is a meronym (and these
%%       are its holonyms)

word("bumper").
meronyms(W,SW,H,SH) :- &wordnet[08,W](1,SW,H,SH),word(W).

meronyms("bumper",2,"machine",1), 
meronyms("bumper",2,"automobile",1), 
meronyms("bumper",2,"car",1)?
