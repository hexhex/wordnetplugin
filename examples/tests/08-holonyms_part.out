{meronyms("bumper",2,"motortruck",2), meronyms("bumper",2,"truck",2), meronyms("bumper",2,"motorcar",1), meronyms("bumper",2,"machine",1), meronyms("bumper",2,"automobile",1), meronyms("bumper",2,"auto",1), meronyms("bumper",2,"car",1), word("bumper")}
